cdap-tms/src/main/java/co/cask/cdap/messaging/store/leveldb/LevelDBTableFactory.java

```diff
@@ -99,4 +114,26 @@ private File ensureDirExists(File dir) throws IOException {
     }
     return dir;
   }
+
+  private class DataCleanup implements Runnable {
+
+    @Override
+    public void run() {
+      if (metadataTable == null || payloadTable == null || messageTable == null) {
+        return;
+      }
+
+      long timeStamp = System.currentTimeMillis();
+      try {
+        List<TopicId> topicIdList = metadataTable.listTopics();
```

Comment:
Instead of just list existing topics, you can add a method to `LevelDBMetadataTable` to iterates over all topics. Something like:

`CloseableIterator<TopicMetadata> scanTopics(boolean includeDeleted) throws IOException`